### ISSUE TITLE
Differential Equations

### ISSUE THREAD

**Issue by cscherrer** [cscherrer/Soss.jl#1]
Implement SDE, etc using [DifferentialEquations.jl](https://github.com/JuliaDiffEq/DifferentialEquations.jl) and (most likely) Stan-like approach

**Comment by ChrisRackauckas** [comment]
You probably don't want to take on the dependency, so we are building integration with the different backends into [DiffEqBayes.jl](https://github.com/JuliaDiffEq/DiffEqBayes.jl). If you have some starter code that would be great!

**Comment by cscherrer** [comment]
I don't understand how Soss could work as a backend. Can you tell me more about what you have in mind? 

Soss can call any Julia code, so a model could have any DiffEq code as part of it. But I'm open to other ideas

**Comment by ChrisRackauckas** [comment]
>Soss can call any Julia code, so a model could have any DiffEq code as part of it. But I'm open to other ideas

Yes, but that would require learning Soss or knowing Bayesian stats. The point of DiffEqBayes is to make the integration automatic, like with Stan:

```julia
bayesian_result = stan_inference(prob1,t,data,priors)
```

you give it an ODE, the time points with data, the data points, and a collection of priors, and it spits out chains. The problem is structured enough that it's clear exactly what the model should be (or parameters for things like likelihood distribution). There's a Stan and a Turing backend with the same interface.

**Comment by cscherrer** [comment]
Oh, guess I need to look at how they do this. There must be a Stan model specified somewhere, is this just fixed? Does it make sense to have the same model every time?

**Comment by ChrisRackauckas** [comment]
>Does it make sense to have the same model every time?

Yes.

>There must be a Stan model specified somewhere, is this just fixed?

It interpolates in the differential equation, time points, data, and priors. The rest is fixed.

https://github.com/JuliaDiffEq/DiffEqBayes.jl/blob/master/src/stan_inference.jl#L51-L83

**Comment by cscherrer** [comment]
Oh interesting. Yeah that should be pretty easy I think. I was going to say you could write it just as easily by hand, but Soss will eventually make it easier to do things like Posterior predictive checks (described as part of https://github.com/cscherrer/Soss.jl/issues/9).

So... Translate the Stan model you gave, then do Stan-like inference and send you the resulting code?

**Comment by cscherrer** [comment]
Ok cool. Translating is easy. I broke the `logdensity` function in changing the interface all around, but it will be working again soon. A bit tougher is inference, which requires gradients (probably via `XGrad`) and NUTS or a similar sampler (eventually `DynamicHMC`, but I might use the one I wrote temporarily until that but is connected).
